README.md

```diff
@@ -1,129 +1,93 @@
-export REDFISH_USERNAME=admin; export REDFISH_PASSWORD=admin; export REDFISH_HOSTADDR=127.0.0.1
-```
+## Event Subscription to BMC
 
-### Deploy for basic tests
-```shell
-make deploy-basic
-```
+Hardware Event Proxy subscribes to Redfish Events by sending a subscription request to the baseboard management controller (BMC) of the target hardware. The request should include the webhook URL of `Hardware Event Proxy` as the destination address. A perfered way of subscription is via [BMCEventSubscription CRD](https://github.com/metal3-io/metal3-docs/pull/167).
 
-### Undeploy
-```shell
-make undeploy-basic
+```yaml
+apiVersion: metal3.io/v1alpha1
+kind: BMCEventSubscription
+metadata:
+  name: worker-1-events
+spec:
+   hostRef: ostest-worker-1
+   destination: https://hw-event-proxy-cloud-native-events.apps.corp.example.com/webhook
+   context: “SomeUserContext”
+   eventTypes:
+   - Alert
+   protocol: Redfish
+   httpHeadersRef:
+     name: some-secret-name
+     namespace: cloud-native-events
+status:
+  errorMessage: ""
+  errorCount: 0
+  subscriptionID: aa618a32-9335-42bc-a04b-20ddeed13ade
 ```
 
-## End to End Tests
-
-Prerequisite: a working Kubernetes cluster. Have the environment variable `KUBECONFIG` set pointing to your cluster.
-
-### Build Test Tool Image
-```shell
-cd e2e-tests
-make build
-scripts/build-image.sh
-TAG=xxx
-podman push localhost/redfish-event-test:${TAG} quay.io/redhat_emp1/redfish-event-test:latest
+## Subscribe to Hardware Event Proxy
+### Create Subscription with JSON Example
+Request
+```json
+{
+  "Resource": "/cluster/node/nodename/redfish/event",
+  "UriLocation”: “http://localhost:9089/event"
+}
 ```
 
-### Basic Test
-```shell
-make test
+Response
+```json
+{
+  "ID": "da42fb86-819e-47c5-84a3-5512d5a3c732",
+  "Resource": "/cluster/node/nodename/redfish/event",
+  "endpointURI": "http://127.0.0.1:9089/event",
+  "URILocation": "http://localhost:8089/api/cloudNotifications/v1/subscriptions/da42fb86-819e-47c5-84a3-5512d5a3c732"
+}
 ```
-The basic test sets up one test pod and **one** consumer in the same node and sends out Redfish Events to the hw-event-proxy at a rate of 1 msg/sec.
-
-The contents of the received events are verified in the test. The list of fields to check are defined the file [`e2e-tests/data/EVENT_FIELDS_TO_VERIFY`](e2e-tests/data/EVENT_FIELDS_TO_VERIFY).
-
-The events to be tested are defined in the `e2e-tests/data` folder with one JSON file per event. List of events are described [here](e2e-tests/data/README.md).
 
-> 📝 New tests can be added by simply adding JSON files in `e2e-tests/data`.
-
-
-### Performance Test
-```shell
-make test-perf
+### Create Subscription with Golang Example
+```go
+package main
+import (
+	v1pubsub "github.com/redhat-cne/sdk-go/v1/pubsub"
+    v1amqp "github.com/redhat-cne/sdk-go/v1/amqp"
+	"github.com/redhat-cne/sdk-go/pkg/types"
+)
+func main(){
+    
+    resourceAddressHwEvent := "/cluster/node/nodename/redfish/event"
```

Comment:
`nodename` should be real node name form env

Reply:
updated